scraper.py

```diff
@@ -0,0 +1,88 @@
+import requests, re, json
+import dateutil.parser
+from bs4 import BeautifulSoup
+
+url = 'https://krew.info/zapasy/'
+
+
+# ['0 Rh-', '0 Rh+', 'A Rh-', 'A Rh+', 'B Rh-', 'B Rh+', 'AB Rh-', 'AB Rh+']
+def get_all_blood_types(rows):
+    blood_types = []
+    for row in rows:
+        blood_types.append(row.find('strong').text)
+    return blood_types
+
+
+# ['Białystok', 'Bydgoszcz', 'Gdańsk', 'Kalisz', 'Katowice', 'Kielce', 'Kraków', 'Lublin', 'Łódź', 'Olsztyn', 'Opole', 'Poznań', 'Racibórz', 'Radom', 'Rzeszów', 'Słupsk', 'Szczecin', 'Wałbrzych', 'Warszawa', 'Wrocław', 'Zielona Góra']
+def get_all_cities(rows):
+    cities = []
+    for row in rows:
+        imgs = row.find_all('img')
+    for img in imgs:
+        cities.append(img.get('alt'))
+    return cities
+
+
+def map_img_to_blood_type(img):
+    match img:
+        case "img/krew0.png":
+            return "STOP"
+        case "img/krew11.png":
+            return "ALMOST_FULL"
+        case "img/krew1.png":
+            return "OPTIMAL"
+        case "img/krew2.png":
+            return "MODERATE"
+        case "img/krew3.png":
+            return "CRITICAL"
+
+
+def get_bank_status(row):
+    bank_status = []
+    imgs = row.find_all('img')
+    for img in imgs:
+        bank_status.append(map_img_to_blood_type(img.get('src')))
+    return bank_status
+
+
+def get_datetime_modified(soup):
+    datetime_full = re.search(r'\d.+', soup.find(string=re.compile("Aktualizacja stanu:"))).group(0) # most reliable way for now, since this tag has no id
+    datetime_iso8601 = (dateutil.parser.parse(datetime_full)).isoformat() + "+02"
+    return datetime_iso8601
+
+
+def main():
+    response = requests.get(url)
+    response.encoding = 'utf-8'
+
+    soup = BeautifulSoup(response.text, 'html.parser')
+    table = soup.find('table')
+
+    for blood_banks in table.find_all('tbody'):
+        rows = blood_banks.find_all('tr')
+
+    blood_types = get_all_blood_types(rows)
+    cities = get_all_cities(rows)
+
+    bank_status = []
```

Comment:
Nitpicky but imho cleaner if variables are declared at the top of the function (look: `bank_status`, `blood_banks`). Do as you wish.

Reply:
While with `bank_status` I can fully agree (and just moved this list to the top of the `main()` function), I cannot move `blood_banks` dict outside the for each loop as it'd be outside of scope of the `city` variable.

Reply:
Right, I see that now. Ok Thx!